rest/encoder/encoder.go

```diff
@@ -96,8 +96,20 @@ func newEncoder(tag string) *form.Encoder {
 		return []string{fmt.Sprint(time.Time(x.(models.Millis)).UnixMilli())}, nil
 	}, models.Millis{})
 	e.RegisterCustomTypeFunc(func(x any) ([]string, error) {
+		if isDay(time.Time(x.(models.Nanos))) {
+			// endpoints that have nanosecond timestamp query parameters are expected to
+			// also work with date strings if a user wants all data from a specific day
+			return []string{fmt.Sprint(time.Time(x.(models.Nanos)).Format("2006-01-02"))}, nil
+		}
 		return []string{fmt.Sprint(time.Time(x.(models.Nanos)).UnixNano())}, nil
 	}, models.Nanos{})
 
 	return e
 }
+
+func isDay(t time.Time) bool {
+	if t.Hour() != 0 || t.Minute() != 0 || t.Second() != 0 || t.Nanosecond() != 0 {
+		return false
+	}
```

Comment:
Would it be more robust if we truncated to the day and compared like so?

```golang
return t.Truncate(24*time.Hour) == t
```

Here's a [playground](https://play.golang.com/p/PAZRKkZHk8y) demonstrating it.

Reply:
Thanks for the example! Currently, this PR is assuming users want a day if they give us midnight in any timezone (which might be an incorrect assumption). Like for example, does a user want a whole day of crypto trades if they're asking for `2021-07-22 00:00:00 UTC+3`? Or do we only assume this if they use UTC?

Also, this comment just made me realize that we probably can't go with this solution at all. I'm pretty sure if a user wants to do something like `params.TimestampGTE = 1626946200000000000` (2021-07-22 09:30:00) and `params.TimestampLTE = 1626969600000000000` (2021-07-23 16:00:00), it'll be marshaled to `timestamp.gte=2021-7-22&timestamp.lte=2021-7-22` 😭 I can't think of a way to only encode this way for `timestamp.eq`.